lib/simp/cli/lib/track_output.rb

```diff
@@ -0,0 +1,94 @@
+require 'pty'
+require 'timeout'
+require 'facter'
+require 'highline/import'
+
+  # Ensure the puppetserver is running ca on the specified port.
+  # Used ensure the puppetserver service is running.
+  def ensure_running(port = nil)
+    port ||= `puppet config print masterport`.chomp
+
+    begin
+      say "> <%= color('Waiting for puppetserver to accept connections on port #{port}', :cyan) %>"
+      curl_cmd = "curl -sS --cert #{::Utils.puppet_info[:config]['certdir']}/`hostname`.pem --key #{::Utils.puppet_info[:config]['ssldir']}/private_keys/`hostname`.pem -k -H \"Accept: s\" https://localhost:#{port}/production/certificate_revocation_list/ca"
+      say "> INFO: #{curl_cmd}" if @verbose
+      running = (%x{#{curl_cmd} 2>&1} =~ /CRL/)
+      unless running
+        system('puppet resource service puppetserver ensure="running" enable=true > /dev/null 2>&1 &')
+        stages = ["\\",'|','/','-']
+        rest = 0.1
+        timeout = 5
+        Timeout::timeout(timeout*60) {
+          while not running do
+            running = (%x{#{curl_cmd} 2>&1} =~ /CRL/)
+            stages.each{ |x|
+              $stdout.flush
+              print "> #{x}\r"
+              sleep(rest)
+            }
+          end
+        }
+        $stdout.flush
+      end
+    rescue Timeout::Error
+      fail("The Puppet Server did not start within #{timeout} minutes. Please start puppetserver by hand and inspect any issues.")
+    end
+  end
+
+  # Track a running process by following its STDOUT output
+  # Prints a '#' for each line of output
+  # returns -1 if error occured, otherwise the line count if PTY.spawn succeeded
+  def track_output(command, port = nil)
+    say "> INFO: #{command}" if @verbose
```

Comment:
@xxx is an instance variable.  This needs to be a module to mix in if you are going to access instance variables.